packages/dds/tree/docs/Tree Content APIs.md

```diff
@@ -0,0 +1,98 @@
+
+## Low Level APIs
+
+These API surfaces are mainly intended to abstract away concrete details of the implementation.
+They are designed to be able to be stable across future optimizations (such as compressed formats) and compatible with future features (like new FieldKinds).
+
+These APIs exist to support writing the high level APIs.
+They are abstractions to allow these optimizing the tree data representations without breaking or complicating the high level API implementations.
+These abstractions also allow comparing performance and correctness between different data representations.
+
+It is currently undecided if these low level APIs will be part of the package's exported API set: even if they are not exported, keeping them stable should be practical, and ease supporting multiple different high level API versions at once.
+
+### Low Level Data Abstraction
+
+Status: currently implemented as "ITreeCursor".
+
+For performance reasons, it is desirable to support multiple tree representations with a single API to access them.
+This allows future changes to add new more optimized tree formats without breaking APIs.
+
+ChunkedForest is a work in progress leveraging this pattern to provide efficiently forkable compressed tree storage.
+
+Note that the various data-structures which can be traversed using cursor are themselves tree content APIs, however any such format which is mainly intended to be worked with through cursor will not be included in this document.
+
+### Rebaseable Change Representations
+
+Status: currently implemented `TChange` types, which are specific to each `ChangeFamily` implementations.
+The `DefaultChangeset` (which is the one used by [SharedTree](../src/shared-tree/README.md)) is `ModularChangeset` from [modular-schema](../src/feature-libraries/modular-schema/README.md)
+
+This is the lowest level change representation, and must support being rebased over other changes.
+It also must have a stable serialized format and stable behavior (what it does when applied to trees, and how it rebases)
+as long as documents which can contain it are supported (which for the `DefaultChangeset` will be from package release to forever).
+
+Expected to accumulate cruft related to backwards compatibility over time,
+so the production implementation (`ModularChangeset`) is designed to be modular to manage this complexity.
+
+Not intended to be exposed in package API.
+
+### Delta
+
+Status: currently implemented `Delta` and `DeltaVisitor`.
+
+Describes a change to a specific version of a tree.
+Cannot be rebased.
+Does not need stable persisted format.
+Independent of `ChangeFamily` (meaning only a single version is needed, regardless of what editing APIs are added).
+
+Used to update [forest](../src/core/forest/README.md) and `AnchorSet`.
+Can also be used to update anything else that stores a copy of some of the tree data that needs to be kept up to date.
+
+### Low Level Editing
```

Comment:
Is this section awaiting more info?

Reply:
Yes, I forgot to finish that as it needed more research. OOps.

Reply:
Added